Wikipedia/Code/WMFArticleImageGalleryDataSource.m

```diff
@@ -26,7 +27,28 @@ @implementation WMFArticleImageGalleryDataSource
 @dynamic emptyView;
 
 - (instancetype)initWithArticle:(MWKArticle*)article {
-    self = [super wmf_initWithItemsAndReverseIfNeeded:article.images.uniqueLargestVariants];
+    
+    NSArray* images = [article.images.uniqueLargestVariants bk_select:^BOOL (MWKImage* image) {
+        // Keep image if it's the article image even if we can't determine its size - we
+        // always want to show article image as first "lead" image by gallery.
+        if ([article.image isEqualToImage:image]) {
+            return YES;
+        }
+        
+        if (!image.width || !image.height) {
+            // HAX: if this image MWKImage record doesn't have a width and height value (because
+            // it wasn't determined by parsing it from the article HTML's image url) see if the
+            // cache can tell us the size.
+            UIImage* img = [[WMFImageController wmf_appImageCache] imageFromDiskCacheForKey:image.sourceURLString];
```

Comment:
This was a know issue pre 5.0. See MWKImage's `updateWithData:` method. It accounted for such cases slipping between the cracks of the html parse:

    if (!self.width || !self.height) {
        UIImage* img = [UIImage imageWithData:data];
        self.width  = [NSNumber numberWithInt:img.size.width];
        self.height = [NSNumber numberWithInt:img.size.height];
    }

This formerly got invoked when we intercepted the WebView's retrieval of image assets. That only happened *after* we parsed the html, of course, so only occasionally did this happen, but the result was that MWKImage pretty much always had width and height we could check. We *could* do the same somehow with SDImageCache at the time it saves an image - we could see if our MWKImage record has a size and if not we could update it on a background thread.